src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/stores/ResourceFormStore.js

```diff
@@ -87,15 +87,139 @@ export default class ResourceFormStore extends AbstractFormStore implements Form
+                types: remoteTypes,
+            } = remoteSchema[name];
+            const {
+                items: localItems,
+                defaultType: localDefaultType,
+                type: localType,
+                types: localTypes,
+            } = localSchema[name] || {};
+
+            if (remoteType === SECTION_TYPE && remoteItems &&
+                localType === SECTION_TYPE && localItems) {
+                result = this.mergeData(
+                    localItems,
+                    remoteItems,
+                    localData,
+                    remoteData
+                );
+                continue;
+            }
+
+            if (remoteType === SECTION_TYPE && remoteItems) {
+                result = this.mergeData(
+                    localSchema,
+                    remoteItems,
+                    localData,
+                    remoteData
+                );
+                continue;
+            }
+
+            if (localType === SECTION_TYPE && localItems) {
+                result = this.mergeData(
+                    localItems,
+                    remoteSchema,
+                    localData,
+                    remoteData
+                );
+                continue;
+            }
+            if (remoteTypes && localTypes
+                && Object.keys(remoteTypes).length > 0 && Object.keys(localTypes).length > 0
+                && localData[name] && remoteData[name]
+                && isArrayLike(localData[name]) && isArrayLike(remoteData[name])
+            ) {
+                for (let key = 0; key < Math.max(remoteData[name].length, localData[name].length); ++key) {
+                    const remoteChildData = toJS(remoteData[name].length > key ? remoteData[name][key] || {} : {});
+                    const localChildData = toJS(localData[name].length > key ? localData[name][key] || {} : {});
+
+                    const localChildDataType = localChildData?.type;
+                    const resultType = localChildDataType && localChildDataType in remoteTypes
+                        ? localChildDataType
+                        : remoteChildData?.type || remoteDefaultType;
+
+                    const localChildSchema =
+                        // $FlowFixMe
+                        localTypes[localChildData.type]?.form || localTypes[localDefaultType].form;
+
+                    const remoteChildSchema = remoteTypes[resultType].form;
+
+                    const resultChildData = this.mergeData(
+                        localChildSchema,
+                        remoteChildSchema,
+                        localChildData,
+                        remoteChildData
+                    );
+
+                    if (!result[name]) {
+                        result[name] = [];
+                    }
+
+                    if (Object.keys(resultChildData).length > 0) {
+                        resultChildData.type = resultType;
+
+                        if (resultChildData.settings) {
```

Comment:
i think this condition should be negated, right?

Reply:
I think the condition can be removed completely, the `settings` are never set before